Lib/test/test_tstring.py

```diff
@@ -369,24 +369,37 @@ def test_ast_structure(self):
 
     def test_error_conditions(self):
         # Test syntax errors
-        with self.assertRaisesRegex(SyntaxError, "'{' was never closed"):
-            eval("t'{")
-
-        with self.assertRaisesRegex(SyntaxError, "t-string: expecting '}'"):
-            eval("t'{a'")
-
-        with self.assertRaisesRegex(SyntaxError, "t-string: single '}' is not allowed"):
-            eval("t'}'")
+        for case, expected_error in (
+            ("t'", "unterminated t-string literal"),
+            ("t'''", "unterminated triple-quoted t-string literal"),
+            ("t''''", "unterminated triple-quoted t-string literal"),
+            ("t'{", "'{' was never closed"),
+            ("t'{'", "t-string: expecting '}'"),
+            ("t'{a'", "t-string: expecting '}'"),
+            ("t'}'", "t-string: single '}' is not allowed"),
+            ("t'{}'", "t-string: valid expression required before '}'"),
+            ("t'{=x}'", "t-string: valid expression required before '='"),
+            ("t'{!x}'", "t-string: valid expression required before '!'"),
+            ("t'{:x}'", "t-string: valid expression required before ':'"),
+            ("t'{x;y}'", "t-string: expecting '=', or '!', or ':', or '}'"),
+            ("t'{x=y}'", "t-string: expecting '!', or ':', or '}'"),
+            ("t'{x!s!}'", "t-string: expecting ':' or '}'"),
+            ("t'{x!s:'", "t-string: expecting '}', or format specs"),
+            ("t'{x!}'", "t-string: missing conversion character"),
+            ("t'{x=!}'", "t-string: missing conversion character"),
+            ("t'{x!z}'", "t-string: invalid conversion character 'z': "
+                         "expected 's', 'r', or 'a'"),
+            ("t'{lambda:1}'", "t-string: lambda expressions are not allowed "
+                              "without parentheses"),
+            ("t'{x:{;}}'", "t-string: expecting a valid expression after '{'"),
+        ):
+            with self.assertRaisesRegex(SyntaxError, expected_error):
+                eval(case)
```

Comment:
Maybe we can move these into a separate `test_syntax_error_messages` test that also calls `self.subTest` and rename this test to something like `test_runtime_errors` (which we can populate with more tests later)?

Reply:
Done